### PR TITLE
Provide dav setting for user's default calendar

### PR DESCRIPTION
**Pull request by tcitworld**
And add tests to handle schedule-default-calendar-URL

Part 1 for #19826

### CODE REVIEW

**Comment by georgehrke** [comment]
Just double checked to make sure:
The dav property is a protected one: https://github.com/nextcloud/3rdparty/blob/master/sabre/dav/lib/CalDAV/Schedule/Plugin.php#L129

**Review by georgehrke — APPROVED** [review]

**Comment by georgehrke** [comment]
@tcitworld What is your concept for the endpoint for setting this? 
Some DAV method or just a Controller endpoint?

**Comment by tcitworld** [comment]
Since the property was read-only in Sabre I only thought of it at first as a Controller endpoint. We could indeed do that through a DAV method, but is it worth it anyway? Does macOS/iOS Calendar handle this?

**Comment by georgehrke** [comment]
No, the only two interesting things you can configure via DAV in macOS are default alarms for timed events and allday events:
```
<A:propertyupdate xmlns:A="DAV:">
	<A:set>
		<A:prop>
			<B:default-alarm-vevent-datetime xmlns:B="urn:ietf:params:xml:ns:caldav">
				<![CDATA[BEGIN:VALARM
				X-WR-ALARMUID:7254E71A-A1F6-41F5-A006-1B435D9B78E3
				UID:7254E71A-A1F6-41F5-A006-1B435D9B78E3
				TRIGGER:-PT30M
				ATTACH;VALUE=URI:Chord
				ACTION:AUDIO
				END:VALARM
				]]>
			</B:default-alarm-vevent-datetime>
		</A:prop>
	</A:set>
</A:propertyupdate>
```

```
<A:propertyupdate xmlns:A="DAV:">
	<A:set>
		<A:prop>
			<B:default-alarm-vevent-date xmlns:B="urn:ietf:params:xml:ns:caldav">
				<![CDATA[BEGIN:VALARM
				X-WR-ALARMUID:659B19ED-7E2B-427A-9ABA-973206876796
				UID:659B19ED-7E2B-427A-9ABA-973206876796
				TRIGGER:-P1DT15H
				ATTACH;VALUE=URI:Chord
				ACTION:AUDIO
				END:VALARM
				]]>
			</B:default-alarm-vevent-date>
		</A:prop>
	</A:set>
</A:propertyupdate>
```

Changing the default calendar does not result in any CalDAV requests.

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -272,12 +288,13 @@ function propFindDefaultCalendarUrl(PropFind $propFind, INode $node) {
 				}
 
 				if (strpos($principalUrl, 'principals/users') === 0) {
-					$uri = CalDavBackend::PERSONAL_CALENDAR_URI;
-					$displayname = CalDavBackend::PERSONAL_CALENDAR_NAME;
+					list(, $UID) = split($principalUrl);
```
Comment (nickvergessen) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php]: No need to scream :scream: 

```suggestion
					list(, $userId) = split($principalUrl);
```

**Review by nickvergessen — APPROVED** [review]
Looks good in general

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 2]
```diff
@@ -272,12 +288,13 @@ function propFindDefaultCalendarUrl(PropFind $propFind, INode $node) {
...
+					$uri = $this->config->getUserValue($UID, 'dav', 'defaultCalendar', CalDavBackend::PERSONAL_CALENDAR_URI);
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php]:
```suggestion
					$uri = $this->config->getUserValue($userId, 'dav', 'defaultCalendar', CalDavBackend::PERSONAL_CALENDAR_URI);
```

**Comment by skjnldsv** [comment]
Conflicts

**Comment by tcitworld** [comment]
Rebased and php-cs-fixed :)

**Pull request merged**